ckanext/protected_resources/plugin.py

```diff
@@ -142,8 +115,15 @@ def protected_resource_unlock(context, data_dict):
 
     resource = p.toolkit.get_action('resource_show')(context, {'id': data_dict['resource_id']})
 
-    if 'is_protected' in resource:
-        del resource['is_protected']
+    if not resource_has_protected_status(data_dict['resource_id']):
+        raise p.toolkit.ValidationError(['This resource is already unlocked'])
+    else:
+        log.error("WE're on the right track my dude")
+        qry = "DELETE FROM resource_protected WHERE resource_id = :resource_id"
+        model.Session.execute(qry, {
+            'resource_id': data_dict['resource_id']
+        })
+        model.Session.commit()
 
     updated_resource = p.toolkit.get_action('resource_update')(context, resource)
```

Comment:
Why do you need an updated_resource for the `unlock` method but not the `lock` method?

Reply:
I don't this can be removed now. Will fix. Nice catch.